src/setup-java.ts

```diff
@@ -12,6 +13,29 @@ async function run() {
       version = core.getInput(constants.INPUT_JAVA_VERSION, {required: true});
     }
 
+    const mvnOpts: MavenOpts = {
+      caCert: core.getInput(constants.INPUT_MAVEN_CA_CERT_B64),
+      keystore: core.getInput(constants.INPUT_MAVEN_KEYSTORE_P12_B64),
+      password: core.getInput(constants.INPUT_MAVEN_KEYSTORE_PASSWORD),
+      settings: core.getInput(constants.INPUT_MAVEN_SETTINGS_B64),
+      securitySettings: core.getInput(
+        constants.INPUT_MAVEN_SECURITY_SETTINGS_B64
+      )
+    };
+
+    if (
+      (mvnOpts.caCert !== '' ||
+        mvnOpts.keystore !== '' ||
+        mvnOpts.password !== '' ||
+        mvnOpts.securitySettings !== '',
+      mvnOpts.settings !== '') &&
+      !validateOptions(mvnOpts)
```

Comment:
I would cram all of that into a function that reads like this:

```suggestion
      isValidOptions(mvnOpts)
```